src/stepFunctions/commands/visualizeStateMachine/aslVisualizationManager.ts

```diff
@@ -33,7 +33,20 @@ export class AslVisualizationManager extends AbstractAslVisualizationManager {
 
         // Existing visualization does not exist, construct new visualization
         try {
-            await this.cache.updateCache(globalStorage)
+            try {
+                await this.cache.updateCache(globalStorage)
+            } catch (err) {
+                // So we can't update the cache, but can we use an existing on disk version.
+                try {
+                    logger.warn(
+                        'Updating State Machine Graph Visualisation assets failed, checking for fallback local cache.'
+                    )
+                    await this.cache.confirmCacheExists()
+                } catch (err) {
+                    logger.error('No local cached State Machine Graph Visualization assets found.')
+                    throw err
+                }
+            }
```

Comment:
Should we move this logic to `StateMachineGraphCache.updateCache` and tweak the log messages a bit? That way `AslVisualizationCDKManager` would benefit from this fix too. Or we could add a new method to `AbstractAslVisualizationManager` and call that instead of `this.cache.updateCache`

Reply:
Hi @JadenSimon , thanks for the review.
I'm very happy to move the logic, my concern was messing up the return meaning of `
updateCache` i.e. it would return positive/no errors even though the cache wasn't updated.

I'll have a look at `AbstractAslVisualizationManager` as hopefully a better pathway.

Reply:
Makes sense to me. `AbstractAslVisualizationManager` is a better place if we don't want to change semantics of existing code.